microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java

```diff
@@ -125,8 +125,17 @@ public void didClose(DidCloseTextDocumentParams params) {
 
 	@Override
 	public void didSave(DidSaveTextDocumentParams params) {
+		// Since the beginning of the MP LS project,
+		// saving a Java file triggers validation for all open Java files,
+		// which can negatively impact performance.
+		// As there's no comment explaining the reason for this behavior,
+		// we are disabling this feature.
+
 		// validate all opened java files which belong to a MicroProfile project
-		triggerValidationForAll(null);
+		// triggerValidationForAll(null);
+		
+		// Validate only the opened Java file
+		validate(documents.onDidSaveTextDocument(params), true);

```

Comment:
@rgrunber it was here that trigger validation for all opened Java files was done. I disable that because I don't see some usecase which requires that. I prefer loosing a not commons usecase (and which usecase?) to have better performance.

Reply:
And I wonder if we should not trigger validation on didSave 

I dont see any value to retrigger validation on didSave 

@datho7561 @rgrunber what do you think about that?

Reply:
Make sense to me